services/QuillCMS/spec/models/graded_response_spec.rb

```diff
@@ -17,10 +17,12 @@
 
     it 'should return response objects for queries' do
       GradedResponse.refresh
-      graded_responses = GradedResponse.where(question_uid: '123').sort_by(&:id)
+      responses = GradedResponse.where(question_uid: '123').sort_by(&:id)
+      response_ids = responses.map(&:id).sort
+      graded_ids = [optimal.id, nonoptimal.id].sort
 
-      expect(graded_responses.first.id).to be response_optimal.id
-      expect(graded_responses.second.id).to be response_nonoptimal.id
+      expect(response_ids).to eq graded_ids
+      expect(responses.first.attributes.keys.sort).to eq nonoptimal.attributes.keys.sort
```

Comment:
I'm not I understand the purpose of this expectation.

Reply:
This is confirmation that the `Response` model json and the `GradedResponse` model json have the exact same keys. If an attribute in the `Response` model changes, say we add a column via a migration, this test would fail, marking that we need to update the `GradedResponse` model, via and equivalent migration.

Reply:
Also, this may not be obvious, but the columns/attributes of the materialized view are locked after the view migration, i.e. you can't do `responses.*` in the query and expect new columns to come in when the original table is migrated. You also have to migrate the view.

https://github.com/scenic-views/scenic#faqs

Reply:
Thanks for the explanation.  I was unaware of that.   I think it's worth documenting that in its own spec:

```
it 'should have the same keys as Responses table' do
  GradedResponse.refresh
  responses = GradedResponse.where(question_uid: '123').sort_by(&:id)
  expect(responses.first.attributes.keys.sort).to eq graded_nonoptimal.attributes.keys.sort
end
```
and then add  same check for `multiple_choice_response_spec.rb` too?